### ISSUE TITLE
Unable to Expose Prometheus Metrics at Port 2020 and Node Exporter at Port 2021 At The Same Time

### ISSUE THREAD

**Issue by qgervacio** [fluent/fluent-bit#4071]
## Bug Report

**Describe the bug**

Running both metrics on port 2020 (https://docs.fluentbit.io/manual/administration/monitoring) and port 2021 (https://docs.fluentbit.io/manual/pipeline/outputs/prometheus-exporter) causes the service on port 2020 to stop working. 

**To Reproduce**

Create the ff config file:

```
[SERVICE]
    Log_Level    info
    HTTP_Server  On
    HTTP_Listen  0.0.0.0
    HTTP_PORT    2020

[INPUT]
    Name cpu

[OUTPUT]
    Name  stdout
    Match *

[INPUT]
    name                node_exporter_metrics
    tag                 node_metrics
    scrape_interval     60

[OUTPUT]
    name                prometheus_exporter
    match               node_metrics
    listen              0.0.0.0
    port                2021
```

Run the ff command.

```
docker run -ti \
  -v /proc:/host/proc \
  -v /sys:/host/sys \
  -v <the-absolute-path-dir-of-fluent-bit-conf>:/fluent-bit/etc \
  -p 2021:2021 \
  -p 2020:2020 \
  fluent/fluent-bit:1.8.0 \
  /fluent-bit/bin/fluent-bit \
    -c /fluent-bit/etc/fluent-bit.conf \
    -p path.procfs=/host/proc -p path.sysfs=/host/sys \
    -f 1
```

Output

```
Fluent Bit v1.8.0
* Copyright (C) 2019-2021 The Fluent Bit Authors
* Copyright (C) 2015-2018 Treasure Data
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2021/09/08 06:14:49] [ info] [engine] started (pid=1)
[2021/09/08 06:14:49] [ info] [storage] version=1.1.1, initializing...
[2021/09/08 06:14:49] [ info] [storage] in-memory
[2021/09/08 06:14:49] [ info] [storage] normal synchronization mode, checksum disabled, max_chunks_up=128
[2021/09/08 06:14:49] [ info] [cmetrics] version=0.1.1
[2021/09/08 06:14:49] [ info] [input:node_exporter_metrics:node_exporter_metrics.1] path.procfs = /proc
[2021/09/08 06:14:49] [ info] [input:node_exporter_metrics:node_exporter_metrics.1] path.sysfs  = /sys
[2021/09/08 06:14:49] [ info] [output:prometheus_exporter:prometheus_exporter.1] listening iface=0.0.0.0 tcp_port=2021
����[2021/09/08 06:14:49] [ info] [http_server] listen iface=0.0.0.0 tcp_port=2020
[2021/09/08 06:14:49] [ info] [sp] stream processor started
2021-09-08T06:14:49.491357401Z node_cpu_seconds_total{cpu="0",mode="idle"} = 60.079999999999998
2021-09-08T06:14:49.491357401Z node_cpu_seconds_total{cpu="0",mode="iowait"} = 0.54000000000000004
2021-09-08T06:14:49.491357401Z node_cpu_seconds_total{cpu="0",mode="irq"} = 0
2021-09-08T06:14:49.491357401Z node_cpu_seconds_total{cpu="0",mode="nice"} = 0
<snip>
```

On test..

```
$ curl http://127.0.0.1:2021/metrics
# HELP node_cpu_seconds_total Seconds the CPUs spent in each mode.
# TYPE node_cpu_seconds_total counter
node_cpu_seconds_total{cpu="0",mode="idle"} 5211.8999999999996 1631070311772
node_cpu_seconds_total{cpu="0",mode="iowait"} 0.88 1631070311772
node_cpu_seconds_total{cpu="0",mode="irq"} 0 1631070311772
node_cpu_seconds_total{cpu="0",mode="nice"} 0 1631070311772
<snip>
```

However for port `2020` the service hangs

```
$ curl http://127.0.0.1:2020 -vvv
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to 127.0.0.1 (127.0.0.1) port 2020 (#0)
> GET / HTTP/1.1
> Host: 127.0.0.1:2020
> User-Agent: curl/7.64.1
> Accept: */*
> 

```

- Steps to reproduce the problem:

Pls see **To Reproduce**

**Expected behavior**

Both prometheus endpoints at ports `2020` and `2021` should be accessible.

**Screenshots**

N/A

**Your Environment**
* Version used: `fluent/fluent-bit:1.8.0`
* Configuration: Yes
* Environment name and version (e.g. Kubernetes? What version?): N/A
* Server type and version: N/A
* Operating System and version:
  ```
  ProductName:	macOS
  ProductVersion:	11.5.2
  BuildVersion:	20G95
  ```
* Filters and plugins:
   `node_exporter_metrics` and `prometheus_exporter`

**Additional context**

**Comment by nokute78** [comment]
It seems to be fixed on v1.8.7. 
Could you check it?

**Comment by qgervacio** [comment]
> It seems to be fixed on v1.8.7.
> Could you check it?
> 
> #4101

Tested ok in 1.8.7

Using

```
[SERVICE]
    Log_Level    info
    HTTP_Server  On
    HTTP_Listen  0.0.0.0
    HTTP_PORT    2020

[INPUT]
    Name cpu

[OUTPUT]
    Name  stdout
    Match *

[INPUT]
    name                node_exporter_metrics
    tag                 node_metrics
    scrape_interval     60

[OUTPUT]
    name                prometheus_exporter
    match               node_metrics
    port                3031
```

Output

```
Fluent Bit v1.8.7
* Copyright (C) 2019-2021 The Fluent Bit Authors
* Copyright (C) 2015-2018 Treasure Data
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2021/09/21 06:29:44] [ info] [engine] started (pid=1)
[2021/09/21 06:29:44] [ info] [storage] version=1.1.1, initializing...
[2021/09/21 06:29:44] [ info] [storage] in-memory
[2021/09/21 06:29:44] [ info] [storage] normal synchronization mode, checksum disabled, max_chunks_up=128
[2021/09/21 06:29:44] [ info] [cmetrics] version=0.2.1
[2021/09/21 06:29:44] [ info] [input:node_exporter_metrics:node_exporter_metrics.1] path.procfs = /proc
[2021/09/21 06:29:44] [ info] [input:node_exporter_metrics:node_exporter_metrics.1] path.sysfs  = /sys
[2021/09/21 06:29:44] [ info] [output:prometheus_exporter:prometheus_exporter.1] listening iface=0.0.0.0 tcp_port=3031
[2021/09/21 06:29:44] [ info] [http_server] listen iface=0.0.0.0 tcp_port=2020
[2021/09/21 06:29:44] [ info] [sp] stream processor started
```
